File: domain/src/main/java/com/github/dkurata38/liblio/domain/library/LibraryId.java
package com.github.dkurata38.liblio.domain.library;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

@EqualsAndHashCode(of = {"value"})
@ToString
public class LibraryId {
    @Getter
    private final String value;

    public LibraryId(String value) {
        this.value = value;
    }
}

File: domain/src/main/kotlin/com/github/dkurata38/liblio/domain/library/LibraryId.kt
package com.github.dkurata38.liblio.domain.library

import java.util.*

data class LibraryId(val value: String) {
    companion object {
        @JvmStatic
        fun generate(): LibraryId {
            return LibraryId(UUID.randomUUID().toString())
        }
    }
}
